Memory exhaustion when running on 32bits

Launching `go-sbot.exe` compiled for `windows/arm` (32bits ARMv7) causes a memory exhaustion error on Nmap:

```
PS C:\Users\andre\ssbc\go-ssb\cmd\go-sbot> .\go-sbot.exe
time=2020-05-13T17:43:43.2457731+01:00 caller=metrics.go:87 module=sbot starting=metrics addr=localhost:6078
time=2020-05-13T17:43:43.2897736+01:00 caller=stdlib.go:89 module=stdlib ts="2020/05/13 17:43:43" msg="saved identity @R3i0Wa6YIBKapV/1Z3heGggoW+1UJVkm88/Us0Tk2WA=.ed25519 to C:\\Users\\andre\\.ssb-go\\secret"
level=warn time=2020-05-13T17:43:43.3117725+01:00 caller=new.go:120 module=sbot event="bot init" msg="loading default idx" idx=userFeeds
badger 2020/05/13 17:43:43 INFO: All 0 tables opened in 1ms
time=2020-05-13T17:43:43.4347706+01:00 caller=log.go:28 check=fatal err="sbot: OpenContacts failed: error getting contacts index: db/idx: badger failed to open: Mmap value log file. Path=C:\\Users\\andre\\.ssb-go\\indexes\\contacts\\db\\000000.vlog. Error=MapViewOfFile: Not enough memory resources are available to process this command."
time=2020-05-13T17:43:43.4447724+01:00 caller=recovery.go:63 event=panic location=CheckFatal panicLog=panics\CheckFatal517618535
panicWithStack: wrote panics\CheckFatal517618535
PS C:\Users\andre\ssbc\go-ssb\cmd\go-sbot>
```

Making that change allowed it to run on windows, but I'm receiving an odd error when running `sbotcli hist` which I suspect might be related. The server has been started with `-fatbot`, it crashes with the following message:

```
level=info time=2020-05-15T13:34:51.1970412+01:00 caller=main.go:371 module=sbot event=serving ID="@crlBvVkCRLHD+ev+jbm0qb0G8swzlEohrRnF9jWsDVQ=.ed25519" addr=:8008 version=snapshot build=
level=debug time=2020-05-15T13:35:05.5771604+01:00 caller=rpc.go:382 module=sbot remote="@|@crlBvVkCRLHD+ev+jbm0qb0G8swzlEohrRnF9jWsDVQ=.ed25519" event=serving
level=debug time=2020-05-15T13:35:05.5771604+01:00 caller=rpc.go:333 module=sbot remote="@|@crlBvVkCRLHD+ev+jbm0qb0G8swzlEohrRnF9jWsDVQ=.ed25519" event="got request" reqID=-1 method=createHistoryStream type=source
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x0 pc=0x734271]

goroutine 25 [running]:
go.cryptoscope.co/ssb/plugins/gossip.(*FeedManager).CreateStreamHistory(0x124aa400, 0x9d1c10, 0x1261a620, 0x337f0008, 0x125ce440, 0x12488150, 0x0, 0x2)
        C:/Users/andre/ssbc/go-ssb/plugins/gossip/feed_manager.go:203 +0x1d
go.cryptoscope.co/ssb/plugins/gossip.(*handler).HandleCall(0x124ba9a0, 0x9d1c10, 0x1261a620, 0x125ce3c0, 0x9d4f00, 0x125ce380)
        C:/Users/andre/ssbc/go-ssb/plugins/gossip/handler.go:212 +0x524
go.cryptoscope.co/muxrpc.(*HandlerMux).HandleCall(0x17a80360, 0x9d1c10, 0x1261a620, 0x125ce3c0, 0x9d4f00, 0x125ce380)
        C:/Users/andre/go/pkg/mod/go.cryptoscope.co/muxrpc@v1.5.4-0.20191205134222-b1563255bffa/handler.go:41 +0xdc
go.cryptoscope.co/muxrpc.(*rpc).fetchRequest.func1(0x125ce380, 0x9d1c10, 0x1261a620, 0x125ce3c0)
        C:/Users/andre/go/pkg/mod/go.cryptoscope.co/muxrpc@v1.5.4-0.20191205134222-b1563255bffa/rpc.go:366 +0x4c
created by go.cryptoscope.co/muxrpc.(*rpc).fetchRequest
        C:/Users/andre/go/pkg/mod/go.cryptoscope.co/muxrpc@v1.5.4-0.20191205134222-b1563255bffa/rpc.go:365 +0x184
PS C:\Users\andre\ssbc\go-ssb\cmd\go-sbot>
```

--- a/README.md
+++ b/README.md
@@ -313,6 +313,16 @@ go-sbot &
 sbotcli connect "net:some.ho.st:8008~shs:SomeActuallyValidPubKey="
 ```
 
+### Startup error / no mmio
+
+The badger key-value database defaults to loading some of it's files using [memory-mapped i/o](https://en.wikipedia.org/wiki/Memory-mapped_I/O). If this turns out to be a problem on your target platform, you can use `go build -tags nommio` when building to fall back to standard files, which can be a bit slower but should still be fully functional.
+
+The error can look like this:
+
+```
+badger failed to open: Mmap value log file. Path=C:\\some\\where\\.ssb-go\\indexes\\contacts\\db\\000000.vlog. Error=MapViewOfFile: Not enough memory resources are available to process this command.
+```
+
 ## Stack links
 
 * [secret-handshake](https://secret-handshake.club) key exchange using [secretstream](https://godoc.org/go.cryptoscope.co/secretstream)

--- a/keys.go
+++ b/keys.go
@@ -16,9 +16,6 @@ import (
 	"go.cryptoscope.co/secretstream/secrethandshake"
 )
 
-// SecretPerms are the file permissions for holding SSB secrets.
-const SecretPerms os.FileMode = 0600
-
 type KeyPair struct {
 	Id   *FeedRef
 	Pair secrethandshake.EdKeyPair

--- a/keys_default.go
+++ b/keys_default.go
@@ -1,0 +1,9 @@
+// +build linux freebsd darwin
+
+package ssb
+
+import "os"
+
+// SecretPerms are the file permissions for holding SSB secrets.
+// We expect the file to only be accessable by the owner.
+var SecretPerms = os.FileMode(0600)

--- a/keys_windows.go
+++ b/keys_windows.go
@@ -1,0 +1,11 @@
+// +build windows
+
+package ssb
+
+import (
+	"os"
+)
+
+// SecretPerms are the file permissions for holding SSB secrets.
+// Windows has it's own permission system apart from UNIX (owner, group, others)
+var SecretPerms = os.FileMode(0666)

--- a/repo/badger_default.go
+++ b/repo/badger_default.go
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: MIT
 
-// +build freebsd linux windows darwin,amd64 darwin,386
+// +build !nommio
 
 package repo
 

--- a/repo/badger_ios.go
+++ b/repo/badger_ios.go
@@ -1,7 +1,6 @@
 // SPDX-License-Identifier: MIT
 
-// +build darwin
-// +build arm arm64
+// +build nommio
 
 package repo
 

